core/src/main/scala/chisel3/internal/BiConnect.scala

```diff
@@ -218,6 +219,91 @@ private[chisel3] object BiConnect {
     }
   }
 
+  /** Check whether two aggregates can be bulk connected (<=) in FIRRTL. From the
+    * FIRRTL specification, the following must hold for bulk connection:
+    *
+    *   1. The types of the left-hand and right-hand side expressions must be
+    *       equivalent.
+    *   2. The bit widths of the two expressions must allow for data to always
+    *        flow from a smaller bit width to an equal size or larger bit width.
+    *   3. The flow of the left-hand side expression must be sink or duplex
+    *   4. Either the flow of the right-hand side expression is source or duplex,
+    *      or the right-hand side expression has a passive type.
+    */
+  private[chisel3] def canBulkConnectAggregates(
+    sink:                  Aggregate,
+    source:                Aggregate,
+    sourceInfo:            SourceInfo,
+    connectCompileOptions: CompileOptions,
+    context_mod:           RawModule
+  ): Boolean = {
+
+    // check that the aggregates have the same types
+    val typeCheck = CheckTypes.validConnect(
+      Converter.extractType(sink, sourceInfo),
+      Converter.extractType(source, sourceInfo)
+    )
+    // TODO do we need elementwise check for bundles?
+//    val elemsMatch = if(sink.elements.size == source.elements.size) {
+//      val elemValidConnect = sink.elements.zip(source.elements).map {
+//        case (sink, source) => CheckTypes.validConnect(
+//          Converter.extractType(sink._2, sourceInfo),
+//          Converter.extractType(source._2, sourceInfo))
+//        case _ => false
+//      }
+//      elemValidConnect.foldLeft(true)(_ && _)
+//    } else false
+
+    // do not bulk connect DataViews if they don't both reify to the same type
+    // or if their types do not exist
+    val sinkIsView = isView(sink)
+    val sourceIsView = isView(source)
+    var sourceReified: Aggregate = source
+    var sinkReified:   Aggregate = sink
+    if (sinkIsView || sourceIsView) {
+      val sourceType =
+        if (sourceIsView) reifySingleData(source) match {
+          case Some(data: Aggregate) =>
+            sourceReified = data
+            Converter.extractType(data, sourceInfo)
+          case None => None
+        }
+        else Converter.extractType(source, sourceInfo)
+      val sinkType =
+        if (sinkIsView) reifySingleData(sink) match {
+          case Some(data: Aggregate) =>
+            sinkReified = data
+            Converter.extractType(data, sourceInfo)
+          case None => None
+        }
+        else Converter.extractType(sink, sourceInfo)
+
+      if (sourceType != sinkType || sourceType == None)
+        return false
+    }
```

Comment:
I think all of this view/reification logic is unnecessary if you reify outside of this method.